src/bors/mod.rs

```diff
@@ -101,6 +102,6 @@ pub struct CheckSuite {
 pub struct RepositoryState<Client: RepositoryClient> {
     pub repository: GithubRepoName,
     pub client: Client,
-    pub permissions: ArcSwap<UserPermissions>,
+    pub team_api_client: Arc<TeamApiClient>,
```

Comment:
Any specific reason for why to do this? I liked the design where the state only contained actual state (apart from the GH client, which can't easily be replaced with pure state), rather than dependencies from which it is loading stuff.

Reply:
To provide each handler with access to the team API, I could add the client as an argument to the handler itself, or add it to the RepositoryState. The handlers have one to many parameters now, and it feels like the client and permission share the same purpose in RepositoryState, so I just replace one with another. Do you have any suggestions on how to better handle this?

Reply:
Why should handlers have access to the team API? :thinking: Shouldn't it be enough for them to just read the existing permissions stored in the repo? The only part of the code that needs access to the team API is the periodic refresh process (which could be made optional in the future, e.g. if we add a way to run with local permissions for easier testing on your own repo).

Reply:
Yeah, only the refresh handler needs access to the team API to update the permissions. In that case, would it be better to move the client to the refresh handler and leave the state unchanged?

Reply:
Yeah, I would prefer that :)